### ISSUE TITLE
Error Handling needs to be documented

### ISSUE THREAD

**Issue by edjiang** [stormpath/stormpath-framework-spec#32]
While thinking about and building out error handling for the iOS SDK, I've noticed that documentation on error handling is a bit lacking. Currently, using the reference implementation on the `express-stormpath 2.4.0` app, we can get errors in multiple ways:

```
{
  "error": "Account with that email already exists.  Please choose another email."
}
```

```
{
  "name": "ResourceError",
  "status": 400,
  "code": 7100,
  "userMessage": "Invalid username or password.",
  "developerMessage": "Login attempt failed because the specified password is incorrect.",
  "moreInfo": "http://docs.stormpath.com/errors/7100",
  "message": "HTTP 400, Stormpath 7100 (http://docs.stormpath.com/errors/7100): Login attempt failed because the specified password is incorrect.",
  "error": "invalid_grant"
}
```

The spec currently says for many resources:

```
{
  "errors": [
    {
      "message": "User-friendly error message"
    }
  ]
}
```

@robertjd mentioned that the spec mentions an error array since we could potentially have multiple users (for instance, the registration field could have multiple fields that aren't filled in correctly). We should figure out how to deal with form validation errors since I feel like that's a different error type that wouldn't go through the typical error handling flow. 

TLDR: We need to standardize the form error and be sending API clients an error code so they can handle them properly.

**Comment by omgitstom** [comment]
The HTTP status should denote it is an error:

The body of the response should resemble a limited view of the Stormpath Error:

``` json
{
  "status": 400,
  "message": "Invalid username or password."
}
```

For SDK specific errors, the integrations should make cognitive decisions about what to set for status, code, error message.  These should be captured on the page that hosts this information.

**Comment by edjiang** [comment]
Specifically for @robertjd's `express-stormpath`, another error condition I found when disabling the connection to `api.stormpath.com` (this is on the `/oauth/token` endpoint):

```
{
  "inner": {
    "code": "ENOTFOUND",
    "errno": "ENOTFOUND",
    "syscall": "getaddrinfo",
    "hostname": "api.stormpath.com",
    "host": "api.stormpath.com",
    "port": 443
  }
}
```

**Comment by edjiang** [comment]
Thinking about this more, I think the `/oauth/token` endpoint should just follow the OAuth spec for errors? I'm fine with having a different error object for the OAuth endpoint, and it'd be good to work with standard OAuth clients. 

http://tools.ietf.org/html/draft-ietf-oauth-v2-10#section-4.3

**Comment by robertjd** [comment]
Thanks @edjiang for bringing this up!  Closed by our meeting yesterday, as applied in #34
